### PR TITLE
re-add default mosquitto auth config on setup

### PR DESCRIPTION
**Pull request by mstormi**
some people on the forum expect mosquitto to work out of the box

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: functions/packages.bash [hunk 1]
```diff
@@ -311,6 +317,7 @@ mqtt_setup() {
   fi
 }
 
+
```
Comment (ecdye) [inline functions/packages.bash:320]: Remove extra line.

File: functions/packages.bash [hunk 2]
```diff
@@ -287,18 +289,22 @@ mqtt_setup() {
...
-    if ! grep -qs password_file /etc/mosquitto/passwd /etc/mosquitto/mosquitto.conf; then
-      echo -e "\\npassword_file /etc/mosquitto/passwd\\nallow_anonymous false\\n" >> /etc/mosquitto/mosquitto.conf
+    if ! grep -qs password_file ${mosquittoPasswd} ${mosquittoConf}; then
+      echo -e "\\npassword_file ${mosquittoPasswd}\\nallow_anonymous false\\n" >> ${mosquittoConf}
     fi
-    echo -n "" > /etc/mosquitto/passwd
-    chown "mosquitto:${username:-openhabian}" /etc/mosquitto/passwd
-    chmod 660 /etc/mosquitto/passwd
-    if cond_redirect mosquitto_passwd -b /etc/mosquitto/passwd "$mqttUser" "$mqttPasswd"; then echo "OK"; else echo "FAILED"; return 1; fi
+    echo -n "" > ${mosquittoPasswd}
```
Comment (ecdye) [inline functions/packages.bash:299]: Perhaps a more elegant solution would be to use `touch`?
```suggestion
    touch ${mosquittoPasswd}
```

**Review by ecdye — CHANGES_REQUESTED** [review]
Not sure what the point of this is other than to use variable to point to the files, care to elaborate?

**Comment by mstormi** [comment]
> Not sure what the point of this is other than to use variable to point to the files, care to elaborate?



> Not sure what the point of this is other than to use variable to point to the files, care to elaborate?

[This](https://community.openhab.org/t/oh3-mqtt-setup-and-configuration/111494#h-1a-configure-newer-versions-of-mosquitto-4) essentially has changed since our mosquitto implementation. So you cannot connect MQTT clients with the default config, it's what some beginners stumble across

**Comment by mstormi** [comment]
wonder why BATS failed (on some ztab related line) but this is unrelated to this change

**Comment by ecdye** [comment]
That is a change I made in zram, I'm working on fixing it don't worry about it.

**Review by ecdye — APPROVED** [review]
LGTM

**Pull request merged**